build/Jenkinsfile

```diff
@@ -136,7 +136,7 @@ podTemplate(label: 'jenkins-slave',
             sh 'cd /go/src/github.com/mattermost/mattermost-server && make build BUILD_NUMBER=$BUILD_NUMBER'
         }
         container('mattermost-node') {
-            sh 'cd mattermost-webapp && make build BUILD_NUMBER=$BUILD_NUMBER'
+            sh 'cd mattermost-webapp && curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/`git rev-parse HEAD`/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make build'
```

Comment:
this is big line, then it is best practice to split in multiple lines like
```
container('mattermost-node') {
            sh """
cd mattermost-webapp
curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/`git rev-parse HEAD`/mattermost-webapp.tar.gz
mkdir ./dist
tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make build
            """
        }
```

Reply:
what do you think?

Reply:
I generally agree, but in this case, they can't be split up quite that easily because the short-circuiting of the boolean operators is actually important. If the `curl` command fails, it should skip all of the other commands and fall back to the `make` command.

Reply:
got it, ok makes sense